checkov/terraform/modules/module_utils.py

```diff
@@ -242,7 +242,7 @@ def get_module_name(file_path: TFDefinitionKey) -> str | None:
     if not file_path.tf_source_modules:
         return None
     module_name = file_path.tf_source_modules.name
-    if file_path.tf_source_modules.foreach_idx:
+    if file_path.tf_source_modules.foreach_idx is not None:
```

Comment:
in case of an empty string, we want to get `foreach_or_count = '"' ` or `= ''`?

Reply:
I think none of them, I didn't realize foreach_idx can be string. I changed it so it checks if it is int, otherwise it keeps the previous behavior.